examples/gno.land/p/gov/dao/proposals.gno

```diff
@@ -0,0 +1,57 @@
+package dao
+
+import "std"
+
+type ProposalStatus string
+
+// ACTIVE -> ACCEPTED -> EXECUTION(SUCCEEDED/FAILED)
+// ACTIVE -> NOT ACCEPTED
+var (
+	Active              ProposalStatus = "active"               // proposal is still active
+	Accepted            ProposalStatus = "accepted"             // proposal gathered quorum
+	NotAccepted         ProposalStatus = "not accepted"         // proposal failed to gather quorum
+	ExecutionSuccessful ProposalStatus = "execution successful" // proposal is executed successfully
+	ExecutionFailed     ProposalStatus = "execution failed"     // proposal is failed during execution
+)
+
+func (s ProposalStatus) String() string {
+	return string(s)
+}
+
+// PropStore defines the proposal storage abstraction
+type PropStore interface {
+	// Proposals returns the given paginated proposals
+	Proposals(offset, count uint64) []Proposal
+
+	// ProposalByID returns the proposal associated with
+	// the given ID, if any
+	ProposalByID(id uint64) (Proposal, error)
+
+	// Size returns the number of proposals in
+	// the proposal store
+	Size() int
+}
+
+// Proposal is the single proposal abstraction
+type Proposal interface {
+	// Author returns the author of the proposal
+	Author() std.Address
+
+	// Description returns the description of the proposal
+	Description() string
+
+	// Status returns the status of the proposal
+	Status() ProposalStatus
+
+	// Executor returns the proposal executor
+	Executor() Executor
+
+	// Votes returns the votes of the proposal
+	Votes(offset, count uint64) []Vote
+
+	// VotingStats returns the voting stats of the proposal
+	VotingStats() VotingStats
```

Comment:
This is the only vote-related thing we should keep in a Proposal. We should eventually rename to something less vote-oriented like “Stats()” or “ProgressMetrics()”.

While Votes and VoteByMember should be implemented eventually but not take part of this interface.

Reply:
Dropped all voting fetchers from the proposal as part of:
[873ba9b](https://github.com/gnolang/gno/pull/2581/commits/873ba9b1ecd9638fc18337578720e76fb379ae09)

and renamed:

[6ceb72a](https://github.com/gnolang/gno/pull/2581/commits/6ceb72ad3021cfd6fc5843c0125d5fdc084fb643)

Reply:
>While Votes and VoteByMember should be implemented eventually but not take part of this interface.

I've thought _a lot_ about this, and tried out different variations in code, even going so far as to add in a `VoteStore`. This is data that can be fetched through the indexer, and I can't think of any use case where you'd need this data to be available on-chain through a Realm (you can actually have this, but with the indexer and something that posts to your Realm). Keeping them in the current state where the proposal doesn't actually need to keep track of its votes (as you suggested here), introduces a lot of complexity, given that the `Proposal` is wrapped in `SimpleDAO`